presto-main/src/main/java/com/facebook/presto/execution/QueryManagerConfig.java

```diff
@@ -51,6 +51,7 @@
     private int stageCountWarningThreshold = 50;
     private int maxTotalRunningTaskCount = Integer.MAX_VALUE;
     private int maxQueryRunningTaskCount = Integer.MAX_VALUE;
+    private int maxTotalRunningTaskCountToHaltScheduling = Integer.MAX_VALUE;
```

Comment:
Maybe it makes sense to make this optional, so we don't need a magic number (MAX_VALUE).

Reply:
Alternatively, a negative number would never make sense here, so instead of MAX_VALUE we could simply check if it's non-negative.

Reply:
We use MAX_VALUE for some other defaulted values, e.g. `maxDriversPerTask` :)